x-pack/plugins/lens/public/editor_frame_service/embeddable/embeddable_component.tsx

```diff
@@ -43,11 +51,39 @@ export type TypedLensByValueInput = Omit<LensByValueInput, 'attributes'> & {
     | LensAttributes<'lnsMetric', MetricState>;
 };
 
-export type EmbeddableComponentProps = TypedLensByValueInput | LensByReferenceInput;
+export type EmbeddableComponentProps = (TypedLensByValueInput | LensByReferenceInput) & {
+  withActions?: boolean;
+};
+
+interface PluginsStartDependencies {
+  uiActions: UiActionsStart;
+  embeddable: EmbeddableStart;
+  inspector: InspectorStartContract;
+}
 
-export function getEmbeddableComponent(embeddableStart: EmbeddableStart) {
+export function getEmbeddableComponent(core: CoreStart, plugins: PluginsStartDependencies) {
   return (props: EmbeddableComponentProps) => {
+    const { embeddable: embeddableStart, uiActions, inspector } = plugins;
     const factory = embeddableStart.getEmbeddableFactory('lens')!;
-    return <EmbeddableRenderer factory={factory} input={props} />;
+    // by default hide the time badges. Consumers can always override via disabledActions prop
+    const input = { disabledActions: ['CUSTOM_TIME_RANGE_BADGE'], ...props };
+    const [embeddable, loading, error] = useEmbeddableFactory({ factory, input });
+    const hasActions = props.withActions === true;
+
+    if (!hasActions || loading || error || !embeddable) {
+      return (
+        <EmbeddableRoot embeddable={embeddable} loading={loading} error={error} input={input} />
+      );
+    }
```

Comment:
Maybe not for this PR, but I'm thinking we are implicitly overloading what `<EmbeddableRoot>` should do.

Maybe clearer would be something like this:

```suggestion
    if (loading) {
      return (
        <EmbeddableSpinner />
      );
    }
    
    if (error) {
      return (
        <EmbeddableError {...} />
      );
    }
    
    if (!hasActions || !embeddable) {
      return (
        <EmbeddableRoot embeddable={embeddable} loading={loading} error={error} input={input} />
      );
    }
```

Reply:
But the `EmbeddableRoot` contains already some logic for these kind of states.
Maybe a nicer way to render this could be something like:
```tsx
if(embeddable && hasActions){
  return <EmbeddablePanel .../>
}
return <EmbeddableRoot embeddable={embeddable} loading={loading} error={error} input={input} />
```

Let checks and same result

Reply:
> But the `EmbeddableRoot` contains already some logic for these kind of states.

That is my point as well, I am saying maybe it should not contain those.

1. It might be useful for `<EmbeddableRoot>` component itself, if those special cases are separate.
2. Those would be useful for `<EmbeddablePanel>` component, as now `<EmbeddablePanel>` relies on `<EmbeddableRoot>` to render, for example, the error state.